File: 1/2021.07.19/147.Insertion Sort List/java/solution.java
class Solution {
    public ListNode insertionSortList(ListNode head) {
        ListNode dummy = new ListNode();
        while (head != null) {
            ListNode pre = dummy;
            while (pre.next != null) {
                if (pre.next.val <= head.val) {
                    pre = pre.next;
                } else {
                    break;
                }
            }
            ListNode next = head.next;
            head.next = pre.next;
            pre.next = head;
            head = next;
        }
        return dummy.next;
    }
}

File: 1/2021.07.19/147.Insertion Sort List/python/solution.py
class Solution:
    def insertionSortList(self, head: ListNode) -> ListNode:
        dummy = ListNode()
        while head:
            pre = dummy
            while pre.next:
                if pre.next.val <= head.val:
                    pre = pre.next
                else:
                    break
            nex = head.next
            head.next = pre.next
            pre.next = head
            head = nex
        return dummy.next
